### ISSUE TITLE
Open ports requirements

### ISSUE THREAD

**Issue by erottier** [MicrosoftDocs/windows-itpro-docs#11774]
I would like to see a list of ports/applications that has to be opened for this to work.
If I'm correct (which I'm _not_ confident about) then it looks like at least TCP 389 (LDAP) and TCP 88 (Kerberos) should be open for Cloud Trust to work. And/or probably LDAPS (TCP 636) if you use it, but again, not confident about this one.

LDAP for the DCLocator check and Kerberos for the exchange of the partial ticket to the full ticket/TGT.

My sources are:
- https://techcommunity.microsoft.com/t5/windows-it-pro-blog/windows-hello-for-business-hybrid-cloud-kerberos-trust-is-now/ba-p/3651049
- https://cloudbrothers.info/windows-business-cloud-trust-kdc-proxy/
- https://learn.microsoft.com/en-us/windows/security/identity-protection/hello-for-business/hello-how-it-works-authentication?WT.mc_id=AZ-MVP-5004810#microsoft-entra-join-authentication-to-active-directory-using-cloud-kerberos-trust

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 046ca470-8477-5ff1-7021-ad34489ed592
* Version Independent ID: 046ca470-8477-5ff1-7021-ad34489ed592
* Content: [Windows Hello for Business cloud Kerberos trust clients configuration and enrollment - Windows Security](https://learn.microsoft.com/en-us/windows/security/identity-protection/hello-for-business/hello-hybrid-cloud-kerberos-trust-provision?tabs=intune)
* Content Source: [windows/security/identity-protection/hello-for-business/hello-hybrid-cloud-kerberos-trust-provision.md](https://github.com/MicrosoftDocs/windows-itpro-docs/blob/public/windows/security/identity-protection/hello-for-business/hello-hybrid-cloud-kerberos-trust-provision.md)
* Product: **windows-client**
* Technology: **itpro-security**
* GitHub Login: @paolomatarazzo
* Microsoft Alias: **paoloma**

**Comment by paolomatarazzo** [comment]
Hello, for a full list of ports required by AD DS, please review this article: https://learn.microsoft.com/troubleshoot/windows-server/identity/config-firewall-for-ad-domains-and-trusts

**Comment by erottier** [comment]
This is NOT a solution. This is frankly lazy and an too easy workaround. Pointing to a general list of all possible ports which a domain uses is not good or secure.

There should be a minimum required set of applications\ports listed for the specified solutions. Like: Cloud trust requires LDAP (secure) and Kerberos open (line of sight) from the WHfB client to the DCs.

This is much more secure then just dumping all ports on open, like SMB (445) or RPC (135), which are insecure _when not needed_.

**Comment by paolomatarazzo** [comment]
@erottier I pointed out to that article because there are no differences in terms of port requirements, otherwise they would have been called out in the article. I think *it IS* a solution. When you ask which ports to open, I assume you are asking from a server standpoint (DC)? In which case, the article I provided is accurate. If you are asking from a client perspective, you don't need to open any specific ports. The client requires to communicate to a DC using the standard ports: 135 (RPC),389(LDAP),3268(GC),88(Kerberos), 445 (SMB). If you have a cert on your DCs and want to use SSL for LDAP and SSL, then 636 and 3269 must be allowed. The linked article seems correct, but maybe I'm misunderstanding your ask. Also, are you asking for hybrid or Microsoft Entra joined? Since it's common to have a mix, you probably want to keep all those ports open. Please let me know if I misunderstood the ask.

**Comment by erottier** [comment]
@paolomatarazzo No, you assume without asking! Please do take care when closing the issue, you actually understand it first.
@aczechowski 

Example:
We have clients which are Azure-joined only, with a VPN for working at home.
Client trust DO require ports, you are mistaken there! LDAP and Kerberos are required.

Also see the links provided in my first post.
Evidence 1:
https://techcommunity.microsoft.com/t5/windows-it-pro-blog/windows-hello-for-business-hybrid-cloud-kerberos-trust-is-now/ba-p/3651049
_"Client contacts on-premises AD domain controller and exchanges the partial TGT for a full TGT"_

Evidence 2:
https://learn.microsoft.com/en-us/windows/security/identity-protection/hello-for-business/hello-how-it-works-authentication?WT.mc_id=AZ-MVP-5004810#microsoft-entra-join-authentication-to-active-directory-using-cloud-kerberos-trust
A: _"the provider uses the DClocator service to locate a 2016 domain controller."_
B: _"After locating a domain controller, the Kerberos provider sends a partial TGT that it received [...] to the domain controller"_

Using this last example, A = LDAP and B = Kerberos.
Both from the client to the DCs.

I would expect this to be documented in the procedure instead of collecting it from all over without even specifics. There just is not usable list, while it IS required.

**Comment by paolomatarazzo** [comment]
@erottier you are correct, the article is called "Configure and provision Windows Hello for Business - cloud Kerberos trust", not "Authenticate to Active Directory". If you require your clients to authenticate to on-premises AD, then the standard Active Directory ports are required. That is, your clients must be able to reach to a DC using the ports that I already mentioned before (and are documented in a different article). Nothing has changed from a ports perspective simply because you adopt a newer feature. If that was the case, it would be documented in the feature article. I'm not sure in which part I'm mistaken. A Microsoft Entra join device still requires those ports (with exception of SMB, since they don't get GPOs from a DC). Where did you read me mentioning otherwise? None of the WHfB documentation (which has been around for many years) discusses about port requirements for AD auth, simply because it has the same requirements as standard auth. 
Are you asking to add a link to the documentation I provided you with?

I'm confused about this ask, but I hope my explanation clarifies it. If not, let me know.

**Comment by erottier** [comment]
@paolomatarazzo You are incorrect. Those applications are a REQUIREMENT for Cloud kerberos trust. This should be mentioned. It does NOT work with without it.

**Comment by paolomatarazzo** [comment]
If I understand correctly, you are suggesting to document that to authenticate to on-premises resources protected by AD while using cloud Kerberos trust, the standard AD ports are required? I can easily do so, but it's implied (like in all the other WHfB documentation). Take as an example *key trust*... how is it different from a ports perspective?

**Comment by paolomatarazzo** [comment]
Anyways, I don't want to waste your time, I will check internally and see if it makes sense to add ports requirements. Thank you.

**Comment by erottier** [comment]
I'm stating that the documentation should state ALL requirements. Cloud trust needs open ports, it _cannot_ function without it.

Machines with no AD connection and needing to connect to on-prem resources (hench Cloud Trust) _need_ those ports. 
It is definitely not implied. It 'just works' for a lot of people who -very insecurely- just open everything to the DCs.

But it's a very bad practice to open everything up because the documentation lacks the required information.  

Thansk for checking. If there is documentation where it's specified (and not implied - which is practically guesswork) I would love to see it.
